src/core/adapters/personalization.ts

```diff
@@ -0,0 +1,117 @@
+import Actions from '../actions';
+import Configuration from '../configuration';
+import Selectors from '../selectors';
+import Store from '../store';
+import ConfigAdapter from './configuration';
+
+namespace Personalization {
+  type ExtractableAction = Actions.SelectRefinement & Actions.AddRefinement;
+
+  export const DAYS_IN_SECONDS = 86400;
+
+  export const extractBias = (action: ExtractableAction, state: Store.State) => {
+    const config = Selectors.config(state).personalization.realTimeBiasing;
+    const byId = Selectors.realTimeBiasesById(state);
+    const { field, value } = extractRefinement(action, state);
+
+    if (!config.attributes[field]) {
+      return;
+    }
+
+    return {
+      variant: field,
+      key: value,
+      bias: (byId[field] && byId[field][value]) ? {
+        ...byId[field][value],
+        lastUsed: Math.floor(Date.now() / 1000)
+      } : generateNewBias()
+    };
+  };
+
+  // tslint:disable-next-line max-line-length
+  export const extractRefinement = ({ type, payload }: ExtractableAction, state: Store.State): { field: string, value: string } => {
+    switch (type) {
+      case Actions.ADD_REFINEMENT:
+        if (!payload.range) {
+          return { field: payload.navigationId, value: payload.value };
+        }
+        break;
+      case Actions.SELECT_REFINEMENT:
+        return Selectors.refinementCrumb(state, payload.navigationId, payload.index);
+      default:
+    }
+    return { field: undefined, value: undefined };
+  };
+
+  export const generateNewBias = () => ({
+    lastUsed: Math.floor(Date.now() / 1000)
+  });
+
+  // tslint:disable-next-line max-line-length
+  export const transformToBrowser = (state: Store.Personalization.Biasing, reducerKey: string): BrowserStorageState => ({
+    allIds: state.allIds.map(({ variant, key }) => ({
+      variant,
+      key,
+      ...state.byId[variant][key]
+    }))
+  });
+
+  // tslint:disable-next-line max-line-length
+  export const transformFromBrowser = (incomingState: BrowserStorageState, state: Store.State): Store.Personalization.Biasing => {
+    const config = Selectors.config(state);
+    // tslint:disable-next-line max-line-length
+    const olderThanTime = Math.floor(Date.now() / 1000) - DAYS_IN_SECONDS * ConfigAdapter.extractRealTimeBiasingExpiry(config);
+    const filteredIncomingState = incomingState.allIds.filter((element) => element.lastUsed >= olderThanTime);
+    let allIds = [];
+    const byId = {};
+    filteredIncomingState.forEach(({ variant, key, lastUsed }) => {
+      allIds.push({ variant, key });
```

Comment:
rather than pushing everything into `allIds` then pruning them back out, I'd prefer if you only conditionally add them